perf/perf_sdt_probe.py

```diff
@@ -0,0 +1,136 @@
+
+        # Check if libpthread has valid SDT markers
+        new_val = self.run_cmd_out("perf list | grep SDT|wc -l")
+        if val == new_val:
+            self.fail("No SDT markers available in the %s", self.libpthread)
+
+    def remove_library(self):
+        perf_remove = "perf buildid-cache -v --remove %s" % self.libpthread
+        self.is_fail = 0
+        self.run_cmd(perf_remove)
+        if self.is_fail:
+            self.fail("Unable to remove %s from builid-cache", self.libpthread)
+
+    def enable_sdt_marker_probe(self):
+        self.sdt_marker = "sdt_libc:memory_mallopt_mmap_max"
+        if self.sdt_marker not in self.run_cmd_out("perf list"):
+            self.fail("SDT marker %s not available", self.sdt_marker)
+
+        enable_sdt_probe = "perf  probe --add %s" % self.sdt_marker
+        self.is_fail = 0
+        self.run_cmd(enable_sdt_probe)
+        if self.is_fail:
+            self.remove_library()
+            self.fail("Unable to probe SDT marker event %s", self.sdt_marker)
+
+    def disable_sdt_marker_probe(self):
+        disable_sdt_probe = "perf probe --del %s" % self.sdt_marker
+        self.is_fail = 0
+        self.run_cmd(disable_sdt_probe)
+        if self.is_fail:
+            self.remove_library()
+            self.fail("Unable to remove SDT marker event probe %s",
+                      self.sdt_marker)
+
+    def record_sdt_marker_probe(self):
+        record_sdt_probe = "perf record -e %s -aR sleep 1" % self.sdt_marker
+        self.is_fail = 0
+        self.run_cmd(record_sdt_probe)
+        if self.is_fail or not os.path.exists("perf.data"):
+            self.disable_sdt_marker_probe()
+            self.remove_library()
+            self.fail("Perf record of SDT marker event %s failed",
+                      self.sdt_marker)
+
+    def setUp(self):
+        """
+        Setting up the env for SDT markers
+        """
+        smg = SoftwareManager()
+        detected_distro = distro.detect()
+        if 'ppc' not in distro.detect().arch:
+            self.cancel("Test supportd only on  ppc64 arch")
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in detected_distro.name:
+            deps.extend(['libc-dev', 'libc-bin', 'linux-tools-common',
+                        'linux-tools-%s' % platform.uname()[2]])
+        elif detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf', 'glibc', 'glibc-devel'])
+        else:
+            self.cancel("Install the package for perf supported by %s"
+                        % detected_distro.name)
+        for package in deps:
+            if not smg.check_installed(package) and not smg.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def test(self):
+        self.add_library()
+        self.enable_sdt_marker_probe()
+        self.record_sdt_marker_probe()
+        self.disable_sdt_marker_probe()
+        self.remove_library()
```

Comment:
@PraveenPenguin in case there is a failure in enable_sdt_marker_probe(), will the test take care of removing library ?

Reply:
ah i miss that it should be part of tear down , will move that